### PR TITLE
Extend LTP test that it can take url based ltp url

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Added check that user can provide there hosted ltp url

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Please provide a run log or change snippet log.

**Comment by PraveenPenguin** [comment]
job log:

localhost:/home/OpTest/avocado-fvt-wrapper/tests/avocado-misc-tests/generic # avocado run ltp.py -m ltp.py.data/sainity.ltp 
JOB ID     : 30da6727bd3c7776ad3e9856477f20e0d8b492e9
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2038-01-19T19.56-30da672/job.log
 (1/1) ltp.py:LTP.test;run-setup-a005: -
Broadcast message from systemd-journald@localhost (Tue 2038-01-19 20:36:18 EST):

syslogtst[4605]: syslogtst: mail emerg test.

\
Broadcast message from systemd-journald@localhost (Tue 2038-01-19 20:37:36 EST):

syslogtst[4938]: syslogtst: emergency log

FAIL: LTP tests failed: ['crypto_user02', 'fallocate06', 'fanotify09', 'fanotify15'] (3523.66 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2038-01-19T19.56-30da672/results.html
JOB TIME   : 3527.88 s
localhost:/home/OpTest/avocado-fvt-wrapper/tests/avocado-misc-tests/generic #

**Comment by PraveenPenguin** [comment]
@harish-24 added job log file , even added one more commit to as user can provide his custom skip file , please have a look

File: generic/ltp.py [hunk 1]
```diff
@@ -126,11 +133,11 @@ def setUp(self):
     def test(self):
         logfile = os.path.join(self.logdir, 'ltp.log')
         failcmdfile = os.path.join(self.logdir, 'failcmdfile')
-
+        skipfile = self.params.get('skipfile', default='skipfile')
         os.chmod(self.teststmpdir, 0o755)
         self.args += (" -q -p -l %s -C %s -d %s -S %s"
                       % (logfile, failcmdfile, self.teststmpdir,
-                         self.get_data('skipfile')))
+                         self.get_data(skipfile)))
```
Comment (harish-24) [inline generic/ltp.py:140]: This restricts the user to place the file in data directory. I would suggest it not to do that and instead it should take an absolute path

**Review by harish-24 — CHANGES_REQUESTED** [review]

File: generic/ltp.py (hunk 1 above)
Comment (PraveenPenguin) [reply generic/ltp.py:140]: can you suggest the path as i am looking this  for  on use case or if you say can comment in yaml file as it need to be placed in data dir ?

Comment (harish-24) [reply generic/ltp.py:140]: the implementation in script should be to take from data dir by default, and it can optionally be overwritten by user. you get the point?

Comment (PraveenPenguin) [reply generic/ltp.py:140]: no i don't want that way as skipfile name can be changed in my case as want to maintain more skipfile having different set  of test to skip

Comment (harish-24) [reply generic/ltp.py:140]: I am not sure if you follow me here. Considering only one file is going to be used, if the input path can be absolute (be it data directory or any other directory) you can still have different skipfiles as well right? But the existing approach restricts user to have skipfiles only in data directory.

Comment (PraveenPenguin) [reply generic/ltp.py:140]: sorry not very clear to me as i got let me write you are saying we can code skipfile don't get from datadir if it is custom it can be absolute path from system  or http path ?

Comment (harish-24) [reply generic/ltp.py:140]: yes, It can be absolute path from the system is what I meant.

Comment (PraveenPenguin) [reply generic/ltp.py:140]: @harish-24 ah some issue with this approach as if i want to use this in CI i need to have independent logic to place this file in system , so i feel i can go with network path and  using asset fetch skipfile can be populated  is this  approach make sense to you ?

Comment (harish-24) [reply generic/ltp.py:140]: @PraveenPenguin It is up-to you to enhance the test better for requirement. But make sure the existing way of providing does not break as it can be used in other places already.

Comment (PraveenPenguin) [reply generic/ltp.py:140]: @harish-24 sure thanks for review  i will adopt that

**Comment by PraveenPenguin** [comment]
job log:

```
avocado run ltp.py -m ltp.py.data/ltp.yaml 
JOB ID     : 5a298092ee571eac06c5de4462954b7d078cc7df
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-07T03.43-5a29809/job.log
 (1/2) ltp.py:LTP.test;run-setup-general-runltp-power_management_tests-5ef3: PASS (134.93 s)
 (2/2) ltp.py:LTP.test;run-setup-general-runltp-numa-f5b2: PASS (163.07 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-07T03.43-5a29809/results.html
JOB TIME   : 299.26 s
````
relevant log for user defined skipfile 
```
2020-10-07 03:46:10,933 process          L0604 INFO | Running '/var/tmp/avocado_bnfn9gn5/bin/runltp -f power_management_tests -q -p -l /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-07T03.43-5a29809/test-results/1-ltp.py_LTP.test_run-setup-general-runltp-power_management_tests-5ef3/ltp.log -C /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-07T03.43-5a29809/test-results/1-ltp.py_LTP.test_run-setup-general-runltp-power_management_tests-5ef3/failcmdfile -d /var/tmp/avocado_bnfn9gn5 -S /var/tmp/avocado_usgul68x/avocado_job_v_j0j1wn/1-ltp.py_LTP.test_run-setup-general-runltp-power_management_tests-5ef3/skipfile'
````

**Comment by PraveenPenguin** [comment]
@harish-24  pushed changes , please have a look

File: generic/ltp.py.data/ltp.yaml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 setup:
     mem_leak: 0
+    url: 'https://github.com/linux-test-project/ltp/archive/master.zip'
+    skipfileurl: None
```
Comment (harish-24) [inline generic/ltp.py.data/ltp.yaml:4]: this line seems to have a whitespace at the end

File: generic/ltp.py [hunk 3]
```diff
@@ -126,11 +134,20 @@ def setUp(self):
     def test(self):
         logfile = os.path.join(self.logdir, 'ltp.log')
         failcmdfile = os.path.join(self.logdir, 'failcmdfile')
-
+        skipfileurl = self.params.get(
+            'skipfileurl', default=None)
+        if skipfileurl:
+            skipfile = self.fetch_asset(
+                "skipfile", locations=[skipfileurl], expire='7d')
+            shutil.copyfile(skipfile, os.path.join(self.workdir, 'skipfile'))
```
Comment (harish-24) [inline generic/ltp.py:142]: why is this required to be copied? the fetched file is good enough?

File: generic/ltp.py [hunk 4]
```diff
@@ -126,11 +134,20 @@ def setUp(self):
...
+            skipfilepath = os.path.join(self.workdir, 'skipfile')
+        else:
+            skipfilepath = self.get_data('skipfile')
+        self.log.info("mine test %s" % skipfilepath)
```
Comment (harish-24) [inline generic/ltp.py:146]: probably for debug? please remove

**Review by harish-24 — CHANGES_REQUESTED** [review]

File: generic/ltp.py (hunk 4 above)
Comment (PraveenPenguin) [reply generic/ltp.py:146]: ah i miss that

File: generic/ltp.py.data/ltp.yaml (hunk 2 above)
Comment (PraveenPenguin) [reply generic/ltp.py.data/ltp.yaml:4]: sure

**Comment by PraveenPenguin** [comment]
@harish-24 pushed the changes , please have look

**Comment by PraveenPenguin** [comment]
job log :
```
 avocado run ltp.py -m ltp.py.data/ltp.yaml
JOB ID     : 77f0607291a1a64a0940f74054a49889f61cc5f5
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-08T00.43-77f0607/job.log
 (1/2) ltp.py:LTP.test;run-setup-general-runltp-power_management_tests-5ef3: PASS (133.57 s)
 (2/2) ltp.py:LTP.test;run-setup-general-runltp-numa-f5b2: PASS (152.85 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-08T00.43-77f0607/results.html
JOB TIME   : 292.52 s
```

**Review by harish-24 — APPROVED** [review]
LGTM

**Comment by PraveenPenguin** [comment]
@harish-24 if  PR looks good to you can please hit merge button :)

**Comment by harish-24** [comment]
@PraveenPenguin Missed merging this, Thanks. Right-away !
